### PR TITLE
Feature/610 remove default args

### LINKED ISSUE #610: Don't pass default values explicitly to the lifecycle phases
### Description

Currently, we're passing the following flags with their default values to the lifecycle phases: `-layers`, `-platform` and `-app`. There is no reason to do so and it can be confusing to the reader.
We would like to stop passing these values explicitly.

### Proposed solution

Remove the flags `-layers`, `-platform` and `-app` from the calls to the following lifecycle phases: `detect`, `restore`, `analyze`, `build` and `export` (these flags weren't added to the `creator` call - see [this PR](https://github.com/buildpacks/pack/pull/595)).

### Additional context

The phases are located [here](https://github.com/buildpacks/pack/blob/master/internal/build/phases.go).
A description of the lifecycle (creator) flags can be found [here](https://github.com/buildpacks/rfcs/blob/master/text/0026-lifecycle-all.md#usage).

### PR DESCRIPTION
**Pull request by Subhadeep2707**
## Summary

Removed the flags -layers, -platform and -app from the calls to the following lifecycle phases: detect, restore, analyze, build and export.
#### Before
Default values were being passed explicitly for the following flags to the lifecycle phases: -layers, -platform and -app. 
#### After
The flags were removed from the calls to the lifecycle phases.
## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [ ] No

## Related

Resolves #610

### CODE REVIEW

**Comment by Subhadeep2707** [comment]
@jromero I have created a pull request. Please take a look at it.

File: internal/build/phases_test.go [hunk 1]
```diff
@@ -726,9 +721,6 @@ func testPhases(t *testing.T, when spec.G, it spec.S) {
 				h.AssertSliceNotContains(t, configProvider.ContainerConfig().Cmd, "-log-level", "debug")
 				h.AssertIncludeAllExpectedPatterns(t,
 					configProvider.ContainerConfig().Cmd,
-					[]string{"-layers", "/layers"},
-					[]string{"-app", "/workspace"},
-					[]string{"-platform", "/platform"},
 				)
```
Comment (jromero) [inline internal/build/phases_test.go:724]: There are no checks happening here anymore. This assertion can be removed altogether.

File: internal/build/phases.go [hunk 2]
```diff
@@ -23,7 +23,6 @@ const (
 	appDir                    = "/workspace"
```
Comment (yaelharel) [inline internal/build/phases.go:23]: Do we still need it?

Comment (Subhadeep2707) [reply internal/build/phases.go:23]: Thanks for the heads up. Will look into it.

**Comment by jromero** [comment]
Hi @Subhadeep2707,

We have recently made changes to areas that this PR touched which caused a conflict. Feel free to reach out on our [slack](https://slack.buildpacks.io/) if you need any assistance.

**Comment by jromero** [comment]
Hello @Subhadeep2707, it seems like this PR has grown stale. The codebase around this has drastically changed. Since we haven't heard from you in a while we're going to close it out for now. Please feel free to reopen it if you are able to update it.

**Pull request closed without merging**